lib/utils/npm-utils.js

```diff
@@ -86,52 +87,32 @@ function parsePackageName(packageName) {
  * @returns {Object} Gotten peerDependencies. Returns null if npm was not found.
  */
 async function fetchPeerDependencies(packageName) {
-    const npmProcess = spawn.sync(
-        "npm",
-        ["show", "--json", packageName, "peerDependencies"],
-        { encoding: "utf8" }
-    );
+    const { name, version } = parsePackageName(packageName);
 
-    const error = npmProcess.error;
+    try {
+        // eslint-disable-next-line n/no-unsupported-features/node-builtins -- Fallback using built-in fetch
+        const response = await fetch(`https://registry.npmjs.org/${name}`);
+        const data = await response.json();
 
-    if (error && error.code === "ENOENT") {
+        if (data.error) {
+            throw new Error(`Unexpected Error when fetching package \`${name}\`: ${data.error}`);
+        }
 
-        // Fallback to using the npm registry API directly when npm is not available.
-        const { name, version } = parsePackageName(packageName);
-
-        try {
-            // eslint-disable-next-line n/no-unsupported-features/node-builtins -- Fallback using built-in fetch
-            const response = await fetch(`https://registry.npmjs.org/${name}`);
-            const data = await response.json();
-
-            const resolvedVersion =
-        version === "latest" ? data["dist-tags"]?.latest : version;
-            const packageVersion = data.versions[resolvedVersion];
-
-            if (!packageVersion) {
-                throw new Error(
-                    `Version "${version}" not found for package "${name}".`
-                );
-            }
-            return Object.entries(packageVersion.peerDependencies).map(
-                ([pkgName, pkgVersion]) => `${pkgName}@${pkgVersion}`
-            );
-        } catch {
+        const resolvedVersion = semverMaxSatisfying(Object.keys(data.versions), data["dist-tags"]?.[version] ?? version);
```

Comment:
Could you explain why this sentence was modified? What problem does it aim to solve?